evals/features/cve/test_cvss_diff.py

```diff
@@ -0,0 +1,85 @@
+import pytest
+import re
+
+from pydantic_evals import Case, Dataset
+from pydantic_evals.evaluators import EvaluationReason, Evaluator
+
+from aegis.agents import rh_feature_agent
+from aegis.features.cve import CVSSDiffExplainer, CVSSDiffExplainerModel
+
+from evals.features.common import (
+    common_feature_evals,
+    create_llm_judge,
+    handle_eval_report,
+    make_eval_reason,
+)
+
+
+class CVSSDiffCase(Case):
+    def __init__(self, cve_id, has_diff):
+        """cve_id given as CVE-YYYY-NUM is the flaw we rewrite description for."""
+        super().__init__(
+            name=f"cvss-diff-for-{cve_id}",
+            inputs=cve_id,
+            expected_output=has_diff,
+            metadata={"difficulty": "easy"},
+        )
+
+
+# FIXME: the check may be improved/generalized
+def is_cvss_valid(cvss: str) -> bool:
+    return re.match(
+        r"([0-9]+(\.[0-9])?)?[/]?(CVSS:3\.1/AV:./AC:./PR:./UI:./S:./C:./I:./A:.)?", cvss
+    )
+
+
+class CVSSDiffEvaluator(Evaluator[str, CVSSDiffExplainerModel]):
+    async def evaluate(self, ctx) -> EvaluationReason:
+        """check that explanation is provided if and only if CVSS differ"""
+        rh_cvss = ctx.output.redhat_cvss3_score
+        nvd_cvss = ctx.output.nvd_cvss3_score
+        if not is_cvss_valid(rh_cvss) or not is_cvss_valid(nvd_cvss):
+            return make_eval_reason(
+                fail_reason="invalid CVSS score returned by the agent"
+            )
+
+        empty_explanation = len(ctx.output.explanation) == 0
+        cvss_differ = rh_cvss == nvd_cvss
+        return make_eval_reason(
+            empty_explanation == cvss_differ,
+            fail_reason="explanation emptiness does not match CVSS difference",
+        )
+
+
+async def cvss_diff(cve_id: str) -> CVSSDiffExplainerModel:
+    """use rh_feature_agent to rewrite description for the given CVE"""
+    feature = CVSSDiffExplainer(rh_feature_agent)
+    result = await feature.exec(cve_id)
+    return result.output
+
+
+# test cases
+cases = [
+    CVSSDiffCase("CVE-2025-47229", False),
```

Comment:
this comment applies to all features eg. we probably need a few more examples to verify against.